src/pages/workspaces/workspace/Data.js

```diff
@@ -477,6 +477,10 @@ const WorkspaceData = _.flow(
     try {
       setSnapshotDetails(_.set([snapshotName, 'error'], false))
       const entities = await Ajax(signal).Workspaces.workspace(namespace, name).snapshotEntityMetadata(googleProject, snapshotName)
+      //Prevent duplicate id columns
+      Object.keys(entities).forEach(key => {
+        entities[key].attributeNames = entities[key].attributeNames.filter(attr => attr !== entities[key].idName)
+      })
```

Comment:
Could this be

```suggestion
      const entitiesWithoutIds = _.pickBy(value => value.attributeNames !== value.idName, entities)
```

which could also be written as 

```suggestion
      const entitiesWithoutIds = _.pickBy(({attributeNames, idName }) => attributeNames !== idName, entities)
```

?

Note that the reason I suggest [pickBy](https://lodash.com/docs/4.17.15#pickBy) instead of [omitBy](https://lodash.com/docs/4.17.15#omitBy) is that `omit` is noted to be "considerably slower" than `pick` but this may be a bad extrapolation and/or premature optimization as I haven't tested either.

In general, we prefer Lodash FP for the following reasons:

- Lodash handles the case where you try to run the function on an incorrect type. e.g.: null, undefined, Object  (in the case of slice)
  - {}.slice(2,4) // throws an error
  - _.slice(2, 4, {}) // returns an empty array
- When running through your code, if you pulled data in from an outside source and got unexpected garbage, the application will not have an error thrown, and can continue along with the value / type it was expecting when using lodash
- Lodash FP functions support currying. This is especially useful for code changes later. If someone wants to add functionality, it makes it easy to do something like add the function to a flow.
- Lodash FP will not mutate the array. In the case of native slice, it also returns a new array. But there are other functions that can mutate (e.g. push/pop)
- For consistency within the codebase, consistency with how the code operates, and to prevent mutation we tend to prefer using Lodash functions.
- Mixing native and Lodash functions can lead to confusion about behavior.

Reply:
Absolutely!  I'm still a bit unfamiliar with Lodash, and here we're trying to filter an array that is a property of an object for every property in an object but I think I got a version that works.

Reply:
Ah I see…we want to filter each `entity`'s `attributeNames` and not `entity`s themselves as a whole. Sorry for my misleading suggestion.

Reply:
No problem!  I learned some good Lodash.